File: JavaScript Basics/Exams/Exam-09-Jan/02.Parachute/Parachute.js
function solve(arr) {
    var pos = 0;
    //arr.forEach(function(x) { console.log(x) });
    var parRow = 0;
    for (var l = 0; l < arr.length; l++) {
        var firstLine = arr[l].split('');
        for (var k = 0; k < firstLine.length; k++) {
            if (firstLine[k] === 'o') {
                pos = k;
                parRow = l;
            }
        }
    }
    
    for (var j = parRow +1; j < arr.length; j++) {
        var moveLeft = 0;
        var moveRight = 0;
        var ch = arr[j].split('');
        for (var i = 0; i < ch.length; i++) {
            if (ch[i] === '<') {
                moveLeft++;
            } else if (ch[i] === '>') {
                moveRight++;
            }
        }
        pos += (moveRight - moveLeft);
        //console.log(pos);
        if (ch[pos] === '_') {
            //console.log(ch[pos]);
            console.log('Landed on the ground like a boss!');
            console.log(j + ' ' + pos);
            break;
        } 
        else if (ch[pos] === '~') {
            //console.log(ch[pos]);
            console.log('Drowned in the water like a cat!');
            console.log(j + ' ' + pos);
            break;
        } 
        else if (
            ch[pos] === '/' || 
            ch[pos] === '\\' || 
            ch[pos] === '|'
            ) {
            //console.log(ch[pos]);
            console.log('Got smacked on the rock like a dog!');
            console.log(j + ' ' + pos);
            break;
        }
    }

}


//solve([
//    '-------------o-<<--------',
//    '-------->>>>>------------',
//    '---------------->-<---<--',
//    '------<<<<<-------/\\--<--',
//    '--------------<--/-<\\----',
//    '>>--------/\\----/<-<-\\---',
//    '---------/<-\\--/------\\--',
//    '<-------/----\\/--------\\-',
//    '\------/--------------<-\\',
//    '-\___~/------<-----------'
//]);

solve([
    '--o----------------------',
    '>------------------------',
    '>------------------------',
    '>-----------------/\\-----',
    '-----------------/--\\----',
    '>---------/\\----/----\\---',
    '---------/--\\--/------\\--',
    '<-------/----\\/--------\\-',
    '\\------/----------------\\',
    '-\\____/------------------'
]);

File: CSharp Advanced/Exams/Exam-Sample-Problems/16.Parachute/Parachute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _16.Parachute
{
    class Parachute
    {
        static void Main(string[] args)
        {
            var field = new List<string>();

            var line = Console.ReadLine();
            while (!line.Equals("END"))
            {
                field.Add(line);
                line = Console.ReadLine();
            }

            var foundParachutist = false;
            var parachutistX = 0;
            var parachutistY = 0;
            for (int row = 0; row < field.Count; row++)
            {
                for (int col = 0; col < field[row].Length; col++)
                {
                    if (field[row][col].Equals('o'))
                    {
                        parachutistX = row - 1; // It gets ++ down in the code.
                        parachutistY = col;
                        foundParachutist = true;
                        break;
                    }

                    if (foundParachutist)
                    {
                        char currPlace = field[row][col];
                        switch (currPlace)
                        {
                            case '>':
                                parachutistY++;
                                break;
                            case '<':
                                parachutistY--;
                                break;
                        }
                    }
                }

                parachutistX++;

                char nextPlace = field[parachutistX][parachutistY];
                if (nextPlace.Equals('_'))
                {
                    Console.WriteLine("Landed on the ground like a boss!");
                    Console.WriteLine("{0} {1}", parachutistX, parachutistY);
                    break;
                }

                if (nextPlace.Equals('~'))
                {
                    Console.WriteLine("Drowned in the water like a cat!");
                    Console.WriteLine("{0} {1}", parachutistX, parachutistY);
                    break;
                }

                if (nextPlace.Equals('/') ||
                    nextPlace.Equals('\\') ||
                    nextPlace.Equals('|'))
                {
                    Console.WriteLine("Got smacked on the rock like a dog!");
                    Console.WriteLine("{0} {1}", parachutistX, parachutistY);
                    break;
                }
            }
        }
    }
}
